spec/checkers/enum_column_checker/postgresql_spec.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+RSpec.describe DatabaseConsistency::Checkers::EnumColumnChecker, :postgresql do
+  subject(:checker) { described_class.new(model, enum) }
+
+  let(:model) { entity_class }
+  let(:enum) { entity_class.defined_enums.keys.first }
+
+  context 'with enum column' do
+    let(:entity_class) do
+      define_class do |klass|
+        klass.enum field: { value1: 'value1', value2: 'value2' }
+      end
+    end
+
+    before do
+      define_database do
+        create_enum :field_type, %w[value1 value2]
+
+        create_table :entities do |t|
+          t.enum :field, enum_type: 'field_type'
+        end
+      end
+    end
+
+    specify do
+      expect(checker.report).to have_attributes(
+        checker_name: 'EnumColumnChecker',
+        table_or_model_name: entity_class.name,
+        column_or_attribute_name: 'field',
+        status: :ok,
+        error_message: nil,
+        error_slug: nil
+      )
+    end
+  end
+
+  context 'with integer column' do
+    let(:entity_class) do
+      define_class do |klass|
+        klass.enum field: %i[value1 value2]
+      end
+    end
+
+    before do
+      define_database do
+        create_table :entities do |t|
+          t.integer :field
+        end
+      end
+    end
+
+    specify do
+      expect(checker.report).to have_attributes(
+        checker_name: 'EnumColumnChecker',
+        table_or_model_name: entity_class.name,
+        column_or_attribute_name: 'field',
+        status: :fail,
+        error_message: nil,
+        error_slug: :enum_column_type_mismatch
+      )
+    end
+  end
+
+  context 'with string column' do
+    let(:entity_class) do
+      define_class do |klass|
+        klass.enum field: { value1: 'value1', value2: 'value2' }
+      end
+    end
+
+    before do
+      define_database do
+        create_table :entities do |t|
+          t.string :field
+        end
+      end
+    end
+
+    specify do
+      expect(checker.report).to have_attributes(
+        checker_name: 'EnumColumnChecker',
+        table_or_model_name: entity_class.name,
+        column_or_attribute_name: 'field',
+        status: :fail,
+        error_message: nil,
+        error_slug: :enum_column_type_mismatch
+      )
+    end
+  end
+
+  context 'when ActiveRecord below 7' do
```

Comment:
@djezzzl, what do you think about this context?

This is an alternative to the following code from `spec/checkers/enum_type_checker/postgresql_spec.rb`
```ruby
before do
  unless ActiveRecord::VERSION::MAJOR >= 7 && adapter == 'postgresql'
    skip('older versions are not supported with sqlite3')
  end
end
```

I think we should be running tests like these and checking that the checker reports nothing. Otherwise, a bug might be introduced in the `precondition` method, and tests will not catch it.

Reply:
Oh, I see why we need this. CI is running tests against different AR versions. I will revert it.